bin/opam.ml

```diff
@@ -164,19 +164,28 @@ let open_pr ~dry_run ~changes ~remote_repo ~fork_owner ~branch ~token ~title
         | Ok () -> Ok 0
         | Error _ -> msg ())
 
-let parse_remote_repo remote_repo =
-  match Github_repo.from_uri remote_repo with
-  | Some repo -> Ok repo
-  | None ->
-      R.error_msgf
-        "The URL to your remote fork of opam-repository %s does not seem to \
-         point to a github repo.\n\
-         Try editing your config with `dune-release config set remote <URL>` \
-         or providing a valid Github repo URL via the --remote-repo option."
-        remote_repo
-
-let submit ~token ~dry_run ~yes ~opam_repo ~pkgs_to_submit local_repo
-    remote_repo pkgs auto_open ~draft =
+(** Get the value from the [user] config, if it is unset try to parse it from
+    the [remote_repo]. *)
+let get_fork_owner ~fork_owner ~remote_repo =
```

Comment:
Given that this function does only return `fork_owner` when it is set, I think the more sensible thing is to not call it when `fork_owner` is set.

Reply:
Checking on `fork_owner` is the only intent behind this function, to avoid making the larger function even larger. Do you suggest inlining it all ?

Reply:
Yes, because if you pass `fork_owner` you already know the result. No need to call `get_fork_owner` to tell you that the fork owner you had already is indeed the fork owner.

This would be different if there was something to be calculated from the owner so passing in an optional pre-calculated owner was important.